OpenNMT/onmt/Translator.py

```diff
@@ -9,17 +10,34 @@ def __init__(self, opt):
         self.tt = torch.cuda if opt.cuda else torch
 
         checkpoint = torch.load(opt.model)
-        self.model = checkpoint['model']
 
-        self.model.eval()
+        model_opt = checkpoint['opt']
+        self.src_dict = checkpoint['dicts']['src']
+        self.tgt_dict = checkpoint['dicts']['tgt']
+
+        encoder = onmt.Models.Encoder(model_opt, self.src_dict)
+        decoder = onmt.Models.Decoder(model_opt, self.tgt_dict)
+        model = onmt.Models.NMTModel(encoder, decoder)
+
+        generator = nn.Sequential(
+            nn.Linear(model_opt.rnn_size, self.tgt_dict.size()),
+            nn.LogSoftmax())
+
+        model.load_state_dict(checkpoint['model'])
+        generator.load_state_dict(checkpoint['generator'])
 
         if opt.cuda:
-            self.model.cuda()
+            model.cuda()
+            generator.cuda()
         else:
-            self.model.cpu()
+            model.cpu()
+            generator.cpu()
+
+        model.generator = generator
```

Comment:
The main reason I didn't leave generator as a submodule was because both generator and model could be wrapped separately in DataParallel, and I wanted to be able to load state_dicts from models originally trained on X GPUs and continue training them on Y GPUs. Maybe having DataParallel in there wouldn't mess with the state_dicts the way I expected though. I'll double check.